core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequestTests.java

```diff
@@ -69,4 +72,29 @@ public void testSerializationBwc() throws IOException {
             }
         }
     }
+    
+    public void testTopLevelKeys() throws IOException {
+        String createIndex =
+                "{\n"
+                + "  \"FOO_SHOULD_BE_ILLEGAL_HERE\": {\n"
```

Comment:
Typically we would use JsonXContent.contentBuilder() (as can be seen in testSerialization above). Can you please switch to that?